samples/wifi/monitor/src/main.c

```diff
@@ -328,6 +328,57 @@ static int wifi_set_channel(void)
 	return 0;
 }
 
+static int wifi_set_packet_filter(void)
+{
+	int ret;
+	struct net_if *iface;
+	struct wifi_filter_info filter_info = { 0 };
+	uint32_t filter = 0;
+
+	iface = net_if_get_first_wifi();
+	if (iface == NULL) {
+		LOG_ERR("No Wi-Fi interface found");
+		return -1;
+	}
+
+	filter_info.if_index = net_if_get_by_iface(iface);
+
+#if defined(CONFIG_MONITOR_MODE_WIFI_PACKET_FILTER_ALL)
+	filter |= WIFI_PACKET_FILTER_ALL;
+#elif defined(CONFIG_MONITOR_MODE_WIFI_PACKET_FILTER_CUSTOM)
```

Comment:
The below configs would be undefined if `CONFIG_MONITOR_MODE_WIFI_PACKET_FILTER_CUSTOM`, so, we don't need this extra protection.